.github/workflows/osrm-backend.yml

```diff
@@ -275,15 +275,6 @@ jobs:
             CXXCOMPILER: g++-12
             CXXFLAGS: '-Wno-array-bounds -Wno-uninitialized'
 
-          - name: gcc-11-release
```

Comment:
GCC is much more capricious when it comes to std::variant. While with newer GCC versions I was able to make it working, but with GCC 11 it seems it doesn't like constructions like this:

```
struct Array;

using Value = std::variant<Array, ...>;

struct Array {
   std::vector<Value> values;
}
```

Workaround would be to use smth like `boost::recursive_wrapper`, but it would make code more complicated, so I chose the easier way: just drop GCC 11 in favor of GCC 14 I recently added in another PR :) It seems GCC is getting better here: newer versions still had some issues I had no on Clang, but it was at least easy to fix.

Let me please know if there are any objections here.

Reply:
I'd be ok to drop GCC11. I think in the past™️ we had aimed to track the latest Debian release for minimum compilers. No idea which one that is tho. 😉

If you want to go with a recursive wrapper, here's a [write up](https://medium.com/@dennis.luxen/breaking-circular-dependencies-in-recursive-union-types-with-c-17-the-curious-case-of-4ab00cfda10d) of mine from also some time ago on how to implement such a wrapper with like five lines of code in standard C++.